### PR TITLE
[DependencyInjection][HttpKernel] document `Autowire` attribute

### LINKED ISSUE #16625: [DependencyInjection] add `Autowire` parameter attribute
| Q            | A
| ------------ | ---
| Feature PR   | symfony/symfony#45657
| PR author(s) | @kbond
| Merged in    | 6.1

### PR DESCRIPTION
**Pull request by kbond**
Symfony PRs: https://github.com/symfony/symfony/pull/45657 & https://github.com/symfony/symfony/pull/45783
Closes #16625.
Closes https://github.com/symfony/symfony-docs/issues/16636

### CODE REVIEW

**Comment by kbond** [comment]
I believe the "Code Blocks" action will be fixed with https://github.com/symfony/symfony-docs/pull/16627.

File: controller.rst [hunk 1]
```diff
@@ -286,6 +286,45 @@ in your controllers.
...
+:doc:`complex expressions </service_container/expression_language>` can be bound
+to a controller argument with the ``#[Autowire]`` attribute::
+
+    use Psr\Log\LoggerInterface;
+    use Symfony\Component\DependencyInjection\Attribute\Autowire;
+    use Symfony\Component\HttpFoundation\Response;
+    // ...
+
+    /**
+     * @Route("/lucky/number/{max}")
```
Comment (BafS) [inline controller.rst]: Maybe something to do in another PR, but why don't we use the attribute style? Especially with this feature from Symfony 6.1

File: service_container.rst [hunk 2]
```diff
@@ -684,6 +684,56 @@ For a full list of *all* possible services in the container, run:
...
+    // src/Service/MessageGenerator.php
+    namespace App\Service;
+
+    use Psr\Log\LoggerInterface;
+    use Symfony\Component\DependencyInjection\Attribute\Autowire;
+
+    class MessageGenerator
+    {
+        public function __construct(
+            #[Autowire('@monolog.logger.request')] private LoggerInterface $logger
```
Comment (BafS) [inline service_container.rst]:
```suggestion
            #[Autowire('@monolog.logger.request')] private LoggerInterface $logger,
```
to stay consistent with line 731

File: controller.rst (hunk 1 above)
Comment (kbond) [reply controller.rst]: Agree, was just following the method from the other examples in this file. Probably a separate PR is best.

**Review by BafS — APPROVED** [review]

**Comment by wouterj** [comment]
Thank you @kbond. Pretty cool feature, and with great docs :)

I've moved things around, to not duplicate too much. To be honest, the service container articles need some rewriting in general, so the place is not ideal at the moment, but I think having it in the autowiring article makes most sense.
